backend/app/usecase/search/search.go

```diff
@@ -66,13 +69,108 @@ func (s Search) searchResource(resource Resource, orderBy order.Order, query Que
 }
 
 func (s Search) searchShortLink(query Query, orderBy order.Order, filter Filter) (Result, error) {
-	return Result{}, nil
+	if query.User == nil {
+		return Result{}, errors.New("user not provided")
+	}
+	if len(query.Query) == 0 {
+		return Result{}, errors.New("query not provided")
+	}
+
+	shortLinks, err := s.getShortLinkByUser(*query.User)
+	if err != nil {
+		return Result{}, err
+	}
+
+	matchedShortLinks := matchShortLinks(shortLinks, query, orderBy)
+
+	filteredShortLinks := filterShortLinks(matchedShortLinks, filter)
+
+	return Result{
+		shortLinks: filteredShortLinks,
+		users:      nil,
+	}, nil
 }
 
 func (s Search) searchUser(query Query, orderBy order.Order, filter Filter) (Result, error) {
 	return Result{}, nil
 }
 
+func (s Search) getShortLinkByUser(user entity.User) ([]entity.ShortLink, error) {
+	aliases, err := s.userShortLinkRepo.FindAliasesByUser(user)
+	if err != nil {
+		return []entity.ShortLink{}, err
+	}
+
+	return s.shortLinkRepo.GetShortLinksByAliases(aliases)
+}
+
+func matchShortLinks(shortLinks []entity.ShortLink, query Query, orderBy order.Order) []entity.ShortLink {
+	var matchedAliasByAll, matchedAliasByAny, matchedLongLinkByAll, matchedLongLinkByAny []entity.ShortLink
+	keywords := getKeywordsFromQuery(query.Query)
+	for _, shortLink := range shortLinks {
+		if containsAllWords(shortLink.Alias, keywords) {
```

Comment:
Then match and sort can be combine together and be reused across all attributes.

Reply:
I do agree! Maybe we can move this interface and its implementations to the short-d/app repository because it can be applied for repositories different from Short. Wdyt?